### DISCUSSION TITLE
We need a way to filter by license

### DISCUSSION THREAD

**Discussion by dpyy** [civitai/civitai#267] [Ideas]
I can't even figure out which one I can generate images for commercial use and which I cannot

**Comment by mikearts** [comment]
I support it! With licenses in general, there are a lot of incomprehensible things besides.

**Comment by WASasquatch** [comment]
Really, more over, licenses just need to be respected. ***All*** of Stable Diffusion models, and derivatives (fine-tunes, TIs, LoRas, etc) are under Open RAIL-M. People can go ahead and sell the model, etc, but they cannot revoke commercial use, and you do not need to disclose anything. So just ignore Civit AI limitations about commercial usage or credit really if you want to. 

Furthermore, Open RAIL-M license is supposed to ship (distribute) with the models and all derivatives, which they are not. Civit AI should be sending archives that include this license. 

People have even been using licenses which are not even compatible with Open RAIL-M too.

**Reply by dpyy** [reply]
Wait is that actually true? Why did Civit AI create all these specific designations?

**Reply by WASasquatch** [reply]
> Wait is that actually true? Why did Civit AI create all these specific designations?

https://huggingface.co/spaces/CompVis/stable-diffusion-license/blob/main/license.txt

They can't even limit your use as a service, which is also a feature you can revoke on Civit AI. 

Generally you can't add any limitations, or license that contradicts the original license unless explicitly allowed.

**Reply by dpyy** [reply]
But this clause explicitly allows it:

You may add Your own copyright statement to Your modifications and may provide additional or different license terms and conditions - respecting paragraph 4.a. - for use, reproduction, or Distribution of Your modifications, or for any such Derivatives of the Model as a whole, provided Your use, reproduction, and Distribution of the Model otherwise complies with the conditions stated in this License

**Reply by WASasquatch** [reply]
> But this clause explicitly allows it:
> 
> You may add Your own copyright statement to Your modifications and may provide additional or different license terms and conditions - respecting paragraph 4.a. - for use, reproduction, or Distribution of Your modifications, or for any such Derivatives of the Model as a whole, provided Your use, reproduction, and Distribution of the Model otherwise complies with the conditions stated in this License

Did you not read what you just cited, though? 4.a is the entirely of  4 in `Section III: CONDITIONS OF USAGE, DISTRIBUTION AND REDISTRIBUTION`

```
Distribution and Redistribution. You may host for Third Party remote 
access purposes (e.g. software-as-a-service), reproduce and distribute 
copies of the Model or Derivatives of the Model thereof in any medium, 
with or without modifications, provided that You meet the following 
conditions: 
Use-based restrictions as referenced in paragraph 5 MUST be included as 
an enforceable provision by You in any type of legal agreement (e.g. a 
license) governing the use and/or distribution of the Model or 
Derivatives of the Model, and You shall give notice to subsequent users 
You Distribute to, that the Model or Derivatives of the Model are subject 
to paragraph 5. This provision does not apply to the use of Complementary 
Material. 
You must give any Third Party recipients of the Model or Derivatives of 
the Model a copy of this License; 
You must cause any modified files to carry prominent notices stating that 
You changed the files; 
You must retain all copyright, patent, trademark, and attribution notices 
excluding those notices that do not pertain to any part of the Model, 
Derivatives of the Model. 
You may add Your own copyright statement to Your modifications and may 
provide additional or different license terms and conditions - respecting 
paragraph 4.a. - for use, reproduction, or Distribution of Your 
modifications, or for any such Derivatives of the Model as a whole, 
provided Your use, reproduction, and Distribution of the Model otherwise 
complies with the conditions stated in this License. 
```

Which also subsequently includes 5 below this section as well. 

```
5. Use-based restrictions. The restrictions set forth in Attachment A are 
considered Use-based restrictions. Therefore You cannot use the Model and 
the Derivatives of the Model for the specified restricted uses. You may 
use the Model subject to this License, including only for lawful purposes 
and in accordance with the License. Use may include creating any content 
with, finetuning, updating, running, training, evaluating and/or 
reparametrizing the Model. You shall require all of Your users who use 
the Model or a Derivative of the Model to comply with the terms of this 
paragraph (paragraph 5). 
```

**Reply by dpyy** [reply]
yea not a lawyer I can't understand what it's saying. I'll take your word for it but I find it odd then why does citivai have such a feature.

**Reply by WASasquatch** [reply]
> yea not a lawyer I can't understand what it's saying. I'll take your word for it but I find it odd then why does citivai have such a feature.

Licenses are whacky, like this one, it references the past document, and future document, which is just weird. It assumes you somehow have a knowledge of the whole document before even getting there, as it would be negligent to skip around a license just cause it references something, and maintain cohesive comprehension. 

Maybe that's how they seal loop holes in some cases lol

It is very clear though that you can distribute, modify, and host as service any derivatives of 1.4/1.5 which includes fine-tunes as outlined. So someone can sell you a model, but once you have it, it's open season.

**Reply by espritdautomne** [reply]
I would welcome such features myself but i doubt that some authors would agree with such filters as many is trying to capitalized on they models and is certainly not a risk civitai is willing to take.

**Reply by kainatquaderee** [reply]
> > yea not a lawyer I can't understand what it's saying. I'll take your word for it but I find it odd then why does citivai have such a feature.
> 
> Licenses are whacky, like this one, it references the past document, and future document, which is just weird. It assumes you somehow have a knowledge of the whole document before even getting there, as it would be negligent to skip around a license just cause it references something, and maintain cohesive comprehension.
> 
> Maybe that's how they seal loop holes in some cases lol
> 
> It is very clear though that you can distribute, modify, and host as service any derivatives of 1.4/1.5 which includes fine-tunes as outlined. So someone can sell you a model, but once you have it, it's open season.

hello there. can you please confirm that civitai license doesn't matter? can I use sd1.5 models commercially ? or will it sue me? the model even restricted selling of generated images
